kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/proxy/KieServerProxyClient.java

```diff
@@ -108,11 +114,43 @@ public ProxyTarget findTarget(HttpServerExchange exchange) {
     }
 
     @Override
-    public void getConnection(ProxyTarget target, HttpServerExchange exchange, ProxyCallback<ProxyConnection> callback, long timeout, TimeUnit timeUnit) {
+    public void getConnection(ProxyTarget target, HttpServerExchange exchange, final ProxyCallback<ProxyConnection> callback, long timeout, TimeUnit timeUnit) {
         String containerId = containerResolver.resolveContainerId(exchange, configuration.getContainerInfosPerContainer());
         LoadBalancingProxyClient client = containerClients.get(containerId);
-
-        client.getConnection(target, exchange, callback, timeout, timeUnit);
+        try {
+            client.getConnection(target, exchange, new ProxyCallback<ProxyConnection>() {
+                @Override
+                public void completed(HttpServerExchange exchange, ProxyConnection result) {
+                    callback.completed(exchange, result);
+                }
+
+                @Override
+                public void failed(HttpServerExchange httpServerExchange) {
+                    try {
+                        adminHandler.removeUnavailableServer(((CaptureHostLoadBalancingProxyClient) client).getUri());
+                    } finally {
+                        callback.failed(exchange);
+                        ((CaptureHostLoadBalancingProxyClient) client).clear();
+                    }
+                }
+
+                @Override
+                public void couldNotResolveBackend(HttpServerExchange exchange) {
+                    callback.couldNotResolveBackend(exchange);
+                }
+
+                @Override
+                public void queuedRequestFailed(HttpServerExchange exchange) {
+                    callback.queuedRequestFailed(exchange);
+                }
+            }, timeout, timeUnit);
+        } catch (Exception e) {
```

Comment:
What is the point of this try catch statement here? I don't see any checked exception being thrown above.